src/color/color_utils.js

```diff
@@ -252,12 +223,38 @@ p5.ColorUtils.rgbaToHSLA = function(rgba, maxes) {
     }
 
   }
-  return [
-      Math.round(h * 360),
-      Math.round(s * 100),
-      Math.round(l * 100),
-      a * 1
-    ];
+  return [h, s, l, a];
+};
+
+/**
+ * For a color expressed as an hsla array, return the corresponding HSBA value
+ *
+ * @param {Array} hsla An 'array' object that represents a list of HSLA colors
+ * @return {Array} an array of HSBA values
+ */
+p5.ColorUtils.hslaToHSBA = function(hsla) {
+  var h = hsla[0];
+  var s = hsla[1];
+  var l = hsla[2];
+  var a = hsla[3] || 1;
+
+  s *= l < 0.5 ? l : 1 - l;
+  return[ h, 2*s/(l+s), l+s, a];
+};
+
```

Comment:
Should we expand these assignments to be a little less terse and a little more beginner friendly, as we have in the other conversion functions? I am not sure if it matters all the way down here, but it could be nice.